.github/workflows/run-tests.yml

```
@@ -0,0 +1,18 @@
+name: Run Tests
+
+on: [push]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.11'
+      - name: Run tests
+        run: |
+          python -m pip install pipenv
+          python -m pipenv install
+          python -m pipenv run pytest ./tests
```

if we use poetry it would be simple @rbren we should merge it asap because so many changes are made to the pipenv file that cause a lot of conflict

```suggestion
          make build
          poetry run pytest ./tests
```